js/modules/webwindow.mjs

```diff
@@ -1,7 +1,30 @@
 import { httpRequest, createHttpRequest, loadScript, decodeUrl,
          browser, setBatchMode, isBatchMode, isObject, isFunc, isStr, btoa_func } from './core.mjs';
 import { closeCurrentWindow, showProgress, loadOpenui5 } from './gui/utils.mjs';
+import { sha256, sha256_2 } from './base/sha256.mjs';
+
+
+// secret session key used for hashing connections keys
+// only if set, all messages from and to server signed with HMAC hash
+let sessionKey = '';
+
+/** @summary HMAC implementation
+ * @desc see https://en.wikipedia.org/wiki/HMAC for more details
+ * @private */
+function HMAC(key, m, o) {
+   const kbis = sha256(sessionKey + key),
+         ipad = 0x5c, opad = 0x36,
+         ki = [], ko = [];
+   for (let i = 0; i < kbis.length; ++i) {
+      const code = kbis[i];
+      ki.push(code ^ ipad);
+      ko.push(code ^ opad);
+   }
+
+   const hash = sha256_2(ko, (o === undefined) ? m : new Uint8Array(m, o));
 
+   return sha256_2(ki, hash, true);
```

Comment:
Probably, I just swap variable names compare to [wiki HMAC](https://en.wikipedia.org/wiki/HMAC#Definition). 
`kpad` and `ipad` in my implementation are opposite to that is mentioned in wiki. 
I will change my implementation to avoid confusion in the future